config/config.go

```diff
@@ -733,6 +750,49 @@ func (t *OpenTracing) ToTracingConfig() *tracing.Configuration {
 	return ret
 }
 
+// configMutex is used in test to avoid data race in global config.
+type configMutex struct {
```

Comment:
This is not related to this PR, can we remove this? Maybe you can use other PR to do this?

Reply:
Data race caused by the lack of synchronization is obvious now. 

The newly added config option `alter-primary-key` is incompatible with `auto-shard-bits`: when I set `alter-primary-key` to `false` (to test `auto_shard_bits`), any test that altering the primary key is failed with 
```
[ddl:8200]Unsupported add primary key, alter-primary-key is false ("[ddl:8200]Unsupported add primary key, alter-primary-key is false")
```